kernel/src/log_segment.rs

```diff
@@ -68,7 +68,7 @@ impl LogSegment {
         {
             require!(
                 checkpoint_file.version + 1 == commit_file.version,
-                Error::generic(format!(
+                Error::invalid_checkpoint(format!(
```

Comment:
nit: format! is already a String so we can avoid the clone
```suggestion
                Error::InvalidCheckpoint(format!(
```

Reply:
Thanks! That makes sense, let me update this